benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,522 @@
+   * request and receiving the response.
+   */
+  class BlockingUnaryWorker implements Runnable {
+    final BenchmarkServiceGrpc.BenchmarkServiceBlockingStub stub;
+
+    private BlockingUnaryWorker(BenchmarkServiceGrpc.BenchmarkServiceBlockingStub stub) {
+      this.stub = stub;
+    }
+
+    public void run() {
+      while (!shutdown) {
+        long now = System.nanoTime();
+        stub.unaryCall(simpleRequest);
+        delay(System.nanoTime() - now);
+      }
+    }
+  }
+
+  /**
+   * Worker which executes async unary calls. Event timing is the duration between sending the
+   * request and receiving the response.
+   */
+  class AsyncUnaryWorker implements Runnable {
+    final BenchmarkServiceGrpc.BenchmarkServiceStub stub;
+    final Semaphore maxOutstanding = new Semaphore(config.getOutstandingRpcsPerChannel());
+
+    AsyncUnaryWorker(BenchmarkServiceGrpc.BenchmarkServiceStub stub) {
+      this.stub = stub;
+    }
+
+    @Override
+    public void run() {
+      while (!shutdown) {
+        maxOutstanding.acquireUninterruptibly();
+        stub.unaryCall(simpleRequest, new StreamObserver<Messages.SimpleResponse>() {
+          long now = System.nanoTime();
+          @Override
+          public void onNext(Messages.SimpleResponse value) {
+
+          }
+
+          @Override
+          public void onError(Throwable t) {
+
+          }
+
+          @Override
+          public void onCompleted() {
+            delay(System.nanoTime() - now);
+            maxOutstanding.release();
+          }
+        });
+      }
+    }
+  }
+
+  /**
+   * Worker which executes a streaming ping-pong call. Event timing is the duration between
+   * sending the ping and receiving the pong.
+   */
+  class AsyncPingPongWorker implements Runnable {
+    final BenchmarkServiceGrpc.BenchmarkServiceStub stub;
+    final Semaphore maxOutstanding = new Semaphore(config.getOutstandingRpcsPerChannel());
+
+    AsyncPingPongWorker(BenchmarkServiceGrpc.BenchmarkServiceStub stub) {
+      this.stub = stub;
+    }
+
+    @Override
+    public void run() {
+      while (!shutdown) {
+        maxOutstanding.acquireUninterruptibly();
+        final AtomicReference<StreamObserver<Messages.SimpleRequest>> requestObserver =
+            new AtomicReference<StreamObserver<Messages.SimpleRequest>>();
+        requestObserver.set(stub.streamingCall(
+            new StreamObserver<Messages.SimpleResponse>() {
+              long now = System.nanoTime();
+
+              @Override
+              public void onNext(Messages.SimpleResponse value) {
+                delay(System.nanoTime() - now);
+                requestObserver.get().onNext(simpleRequest);
+                now = System.nanoTime();
+                if (shutdown) {
+                  requestObserver.get().onCompleted();
+                }
+              }
+
+              @Override
+              public void onError(Throwable t) {
+
```

Comment:
shouldn't maxOutstanding also be released here and the error should at least be logged?